runtime/server/mcp_server.go

```diff
@@ -364,17 +365,78 @@ Example: Get the top 10 demographic segments (by country, gender, and age group)
 		}
 		defer res.Close()
 
+		// Get the raw response data
 		data, err := res.MarshalJSON()
 		if err != nil {
 			return nil, err
 		}
 
-		return mcp.NewToolResultText(string(data)), nil
+		// Generate open URL for the query
+		openURL, err := s.generateOpenURL(ctx, instanceID, metricsProps)
+		if err != nil {
+			return nil, fmt.Errorf("failed to generate open URL: %w", err)
+		}
+
+		// Parse the response to add the URL
+		response, err := s.addOpenURLToResponse(data, openURL)
+		if err != nil {
+			return nil, fmt.Errorf("failed to add open URL to response: %w", err)
+		}
+
+		return mcp.NewToolResultText(string(response)), nil
 	}
 
 	return tool, handler
 }
 
+// generateOpenURL generates an open URL for the given query parameters
+func (s *Server) generateOpenURL(ctx context.Context, instanceID string, metricsProps map[string]any) (string, error) {
+	// Get instance to access the configured frontend URL
+	instance, err := s.runtime.Instance(ctx, instanceID)
+	if err != nil {
+		return "", fmt.Errorf("failed to get instance: %w", err)
+	}
+
+	// Use the instance's configured frontend URL (which includes org/project and custom domain for cloud, or the correct localhost URL)
+	if instance.FrontendURL == "" {
+		return "", fmt.Errorf("instance %s has no frontend URL configured", instanceID)
+	}
```

Comment:
Probably better to handle this gracefully (i.e. not add an open URL). It might happen in test cases or if people use the MCP server during rollout before a frontend URL has been populated on all the runtimes.

Reply:
Done 👍